### PR TITLE
Abstract auth to the dev feeds. additionally, add pip auth

### PR DESCRIPTION
**Pull request by scbedd**
Context, @turalf is seeing a nightly failure on the `-pr` version of his new package. This is due to the fact that a required nspkg hasn't been deployed to pypi, only the internal feed. 

This nspkg can easily be published to the internal feed through limiting the scope of the build, which WOULD resolve issues going foward. Unfortunately, in our `tox.ini`, the public feed is the only one utilized. This PR adds the necessary auth to make that happen. 

The hilarious parts of python tooling are rearing their heads. Gotta have a `pip.conf` separate from the `.pypirc` which is `twine` related. Thankfully there is an internal `PipAuthenticate` task!

### CODE REVIEW

**Review by praveenkuttappan — APPROVED** [review]

**Comment by scbedd** [comment]
Confirmed! This [absolutely works. ](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=521051&view=results).

File: eng/pipelines/templates/steps/auth_dev_feed.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+parameters:
```
Comment (mitchdenny) [inline eng/pipelines/templates/steps/auth_dev_feed.yml:1]: *nit*: This file name uses ```_``` as a separator, all the other YAML files use ```-```.

**Review by mitchdenny — COMMENTED** [review]
It looks like you are authing the dev feed in the build stage, is this to pull in dependencies that aren't in PyPI yet?

File: eng/pipelines/templates/steps/build-test.yml [hunk 2]
```diff
@@ -32,6 +33,11 @@ steps:
       pip freeze
     displayName: 'Prep Environment'
 
+  - ${{if eq(variables['System.TeamProject'], 'internal') }}:
```
Comment (weshaggard) [inline eng/pipelines/templates/steps/build-test.yml:36]: How does this work for local developers trying to build and test? Do they also need to enable authentication locally to be able to pull from this internal feed during install steps?

Comment (scbedd) [reply eng/pipelines/templates/steps/build-test.yml:36]: They will need to add the authenticate feed to their `pip.conf` or `.pypirc`.

File: eng/pipelines/templates/steps/auth_dev_feed.yml (hunk 1 above)
Comment (scbedd) [reply eng/pipelines/templates/steps/auth_dev_feed.yml:1]: > It looks like you are authing the dev feed in the build stage, is this to pull in dependencies that aren't in PyPI yet?

Absolutely.

Comment (scbedd) [reply eng/pipelines/templates/steps/auth_dev_feed.yml:1]: Done. Thanks for the catch Mitch.

File: eng/pipelines/templates/steps/build-test.yml (hunk 2 above)
Comment (scbedd) [reply eng/pipelines/templates/steps/build-test.yml:36]: I do have a possible alternative implementation. The way that the "auth" works is to create a connection of form

`https://build:***@pkgs.dev.azure.com/azure-sdk/internal/_packaging/azure-sdk-for-python-pr/pypi/simple`.

This is _exactly_ the same form as we'd provide as an argument to pip. Using another --extra-index-url.

There is definitely another way to attack this. I'm honestly fine with either, and was just trying to re-use the same methodology as already used for the publishing.

Comment (weshaggard) [reply eng/pipelines/templates/steps/build-test.yml:36]: After chatting with this I'm find with this approach.

**Review by weshaggard — APPROVED** [review]

**Pull request merged**